improver/utilities/spatial.py

```diff
@@ -463,16 +464,21 @@ def maximum_within_vicinity(self, cube: Cube) -> Cube:
         # points along the edge = 3
         grid_points = (2 * grid_point_radius) + 1
 
+        cube_dtype = cube.data.dtype
+        cube_fill_value = netCDF4.default_fillvals.get(cube_dtype.str[1:])
+        if cube_fill_value is None:
+            cube_fill_value = -np.inf
+
         max_cube = cube.copy()
         unmasked_cube_data = cube.data.copy()
         if np.ma.is_masked(cube.data):
             unmasked_cube_data = cube.data.data.copy()
-            unmasked_cube_data[cube.data.mask] = -np.inf
+            unmasked_cube_data[cube.data.mask] = -cube_fill_value
```

Comment:
I'm wondering, should the right hand side be `cube_fill_value` *not* `-cube_fill_value`? Or should the default fill value of `-np.inf` be `np.inf`? (See also line 481 below.)

Reply:
I think this needs to be a large negative value. This is because we are trying to find the maximum in an area, and we don't want any masked values to be taken into account. If we replace the missing data with a large negative number before taking the maximum the masked values will never be the maximum. These missing values never make it into the output cube, they are just manipulated in the calculation.

Reply:
Many tests fail if this is positive. Needs to be negative.

Reply:
In that case, the default value should be positive infinity, as you are taking the negative on line 476. Thus I think line 470 should be

```python
cube_fill_value = np.inf
```